### PR TITLE
Cirrus: Add dedicated rootless mode testing

### PR DESCRIPTION
**Pull request by cevich**
Signed-off-by: Chris Evich <cevich@redhat.com>

### CODE REVIEW

**Comment by cevich** [comment]
@giuseppe PTAL - very early experiment.  My thinking here is to get a dedicated VM and entry-point first, running as an actual user w/o any sudo/su/runuser/exec trickery.   Then later (as a group) we can figure out the details of "what to test and where".

File: contrib/cirrus/rootless_test.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+record_timestamp "rootless test start"
+
+set -x
+cd "$GOSRC"
+case "${OS_RELEASE_ID}-${OS_RELEASE_VER}" in
+    ubuntu-18) ;&  # Continue to the next item
+    fedora-29) ;&
+    fedora-28) ;&
+    centos-7) ;&
```
Comment (giuseppe) [inline contrib/cirrus/rootless_test.sh]: we can skip centos-7 and rhel-7 for now as rootless doesn't work there

**Comment by giuseppe** [comment]
this is great.  Once it lands we can probably nuke the ad-hoc rootless tests, or pick just a subset of them

**Comment by cevich** [comment]
Hint: See the new "rootless_testing" task among the others :smile:

File: contrib/cirrus/rootless_test.sh (hunk 1 above)
Comment (cevich) [reply contrib/cirrus/rootless_test.sh]: oh okay, that's all just copy-paste.  Initially I was going to only run this vs F29, but it's easy to expand it later as needed.

Comment (cevich) [reply contrib/cirrus/rootless_test.sh]: okay, got it.

**Comment by cevich** [comment]
@giuseppe @baude @mheon PTAL (assuming other tests pass) I think this is ready.  I also added documentation about the task for reference.   At the moment there's no real "test", it just compiles podman as the rootless user.

**Comment by mheon** [comment]
@giuseppe Anything else we need to do here to ensure we have a full subuid/subgid range set up for the user we test as? I'd love to drop the single user rootless container test hack

**Comment by cevich** [comment]
@mheon good point.  At the top-level we're still root, so config. changes can be made as needed to support testing.  General updates to that `setup_rootless()` function should be easy/straight-forward.

Cirrus-CI can also [do things in parallel within a task](https://cirrus-ci.org/guide/writing-tasks/#background-script-instruction),as long as that doesn't hurt our brains too much :smile:   One idea could be to use that to have multiple users running commands at the same time, or have root also doing operations on the system.  Whichever way we want to make life challenging, it's possible :smiley:

**Comment by cevich** [comment]
@mheon I think no need to wait on this.  It's designed as a stub, to be added onto later.

/hold cancel

**Comment by mheon** [comment]
@cevich Fair enough.
LGTM, we can iterate as we go.

**Comment by rhatdan** [comment]
/lgtm

One thing to think about is moving away from fedora29 to fedora 30.  Also would be nice to test on RHEL8 at some point.

**Comment by cevich** [comment]
@rhatdan I'll file jira cards for both.

**Pull request merged**